devito/arch/archinfo.py

```diff
@@ -216,13 +216,36 @@ def lspci_gpu_info(text):
                 gpu_infos.append(gpu_info)
         return filter_real_gpus(gpu_infos)
 
+    def nvidiasmi_gpu_info(text):
+        lines = text.replace('\\n', '\n')
+        lines = lines.splitlines()
+
+        gpu_infos = []
+        for line in lines:
+            gpu_info = {}
+            if 'Off' or 'On' in line:
+                gpu_info = {}
+                match = re.match(r'^\| *[0-9] *(\w*) ([^|]*[^ (...)])[ .]*(Off|On) *'
+                                 r'\| *([^ ]*) *(Off|On)', line)
+                if match:
+                    gpu_info['architecture'] = match.group(1)
+                    gpu_info['product'] = match.group(2)
+                    gpu_info['physicalid'] = match.group(4)
+                    gpu_info['vendor'] = 'NVIDIA'
+                    gpu_infos.append(gpu_info)
+
+        return gpu_infos
+
     # Run homogeneity checks on a list of GPU, return GPU with count if homogeneous,
     #   otherwise None
     def homogenise_gpus(gpu_infos):
         if gpu_infos == []:
             warning('No graphics cards detected')
             return None
 
+        for gpu_info in gpu_infos:
+            gpu_info.pop('physicalid', None)
```

Comment:
Different physical IDs were resulting in None info of GPUs?

Reply:
Can you add a comment?

Reply:
Ok